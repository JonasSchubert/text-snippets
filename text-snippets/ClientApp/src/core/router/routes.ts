export const routes = [
  {
    path: '/',
    name: 'Home',
    component: (): any => import('@/libs/core/views/Home.vue'),
    meta: {
      allowLocaleStorage: true,
      icon: 'mdi-home',
      inNavigationBar: true,
      text: 'message.home'
    }
  },
  {
    path: '/text-snippets',
    name: 'TextSnippets',
    component: (): any => import('@/text-snippets/views/TextSnippets.vue'),
    meta: {
      allowLocaleStorage: true,
      icon: 'mdi-message-processing',
      inNavigationBar: true,
      text: 'message.text-snippet'
    }
  },
  {
    path: '*',
    name: '404',
    component: (): any => import('@/libs/core/views/NotFound.vue'),
    meta: {
      allowLocaleStorage: false,
      inNavigationBar: false
    }
  }
];
