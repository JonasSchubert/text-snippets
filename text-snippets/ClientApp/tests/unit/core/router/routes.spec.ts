import { routes } from '@/core/router/routes';

describe('routes', () => {
  test('should not change unexpected', () => {
    // Assert
    expect(routes).toMatchSnapshot();
  });

  test('should lazy load all but the first component and have defined path and name', () => {
    // Arrange
    for (let index = 0; index < routes.length; index += 1) {
      // Act
      const route = routes[index];

      // Assert
      expect(typeof route.component).toBe('function');
      expect(route.meta).toBeDefined();
      expect(route.name).toBeDefined();
      expect(route.path).toBeDefined();
    }
  });
});
