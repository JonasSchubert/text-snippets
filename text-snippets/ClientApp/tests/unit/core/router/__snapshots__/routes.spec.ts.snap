// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes should not change unexpected 1`] = `
Array [
  Object {
    "component": [Function],
    "meta": Object {
      "allowLocaleStorage": true,
      "icon": "mdi-home",
      "inNavigationBar": true,
      "text": "message.home",
    },
    "name": "Home",
    "path": "/",
  },
  Object {
    "component": [Function],
    "meta": Object {
      "allowLocaleStorage": true,
      "icon": "mdi-message-processing",
      "inNavigationBar": true,
      "text": "message.text-snippet",
    },
    "name": "TextSnippets",
    "path": "/text-snippets",
  },
  Object {
    "component": [Function],
    "meta": Object {
      "allowLocaleStorage": false,
      "inNavigationBar": false,
    },
    "name": "404",
    "path": "*",
  },
]
`;
